roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_master_sg
+
+- name: Create OCP Node Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-node"
+    description: "Labs OCP Node"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 10250
+        to_port: 10250
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_node_sg
+
+- name: Create OCP Infra Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-infra"
+    description: "Labs OCP Infra"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 80
+        to_port: 80
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_infra_sg
+
+- name: Create an Internet Gateway for the new VPC
+  ec2_vpc_igw:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: present
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_gw
+
+- name: Create Gateway subnet route table for the new VPC
+  ec2_vpc_route_table:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    tags:
+      Name: "{{ env_id }}-route-table-gw"
+      env_id: "{{ env_id }}"
+    subnets:
+      - "{{ new_subnets.subnet.id }}"
+    routes:
+      - dest: "0.0.0.0/0"
+        gateway_id: "{{ new_gw.gateway_id }}"
+
+# Create local facts with new subnet id
+- name: Register new created objects as facts
+  set_fact:
+    aws_subnet: "{{ new_subnets.subnet.id }}"
+
+- name: Create Master and etcd instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
+    group: "{{ aws_master_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+      user: sre
+      role: ocp
+      group: masters_aws
+      node_labels: '{"region": "default"}'
+      ocp_role: "{{ master_tag }} {{ etcd_tag}}"
```

Comment:
Can we enhance this to support both combined master/etcd as well as separate etcd ?

Reply:
Sure, but right now there is no option to deploy separate Master(s) and etcd(s) instances, so this will be always true.

But this is a really good topic of discussion; AFAIK OCP 3.7 will require Master and etcd roles to be deployed in different instances, so maybe we could anticipate to this make this change even for 3.6. Thoughts?

Reply:
ok, so maybe it's a follow-up. If so, let's make sure to write an issue to track.

Reply:
Sure